### ISSUE TITLE
App launches on Realme X2 Pro stutter with --release but not --profile

### ISSUE THREAD

**Issue by kpltwr** [flutter/flutter#67879]
Hello,
Everytime, I start the application there is animation stutter. This is mostly with the **release** APK.

Even with the **default counter app**. If I run it with "flutter run --release" command there is animation stutter at start of the application. Seems like frame drops.

Here is a screen recording. No other app is running at the background.

![stutter](https://user-images.githubusercontent.com/23424135/95689966-20048f80-0c32-11eb-94c8-874aec69559d.gif)

**Target Platform:** Android
**Target OS version/browser:** Android 10
**Devices:** Realme X2 Pro

## Logs

$ "flutter analyze"
    found no issues.

$ flutter doctor -v

[✓] Flutter (Channel master, 1.23.0-8.0.pre.284, on Linux, locale en_IN)
    • Flutter version 1.23.0-8.0.pre.284 at /home/kapil/flutter
    • Framework revision eefcff900c (23 hours ago), 2020-10-10 23:42:02 +0200
    • Engine revision 11d756a62e
    • Dart version 2.11.0 (build 2.11.0-207.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/kapil/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /home/kapil/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Android Studio (version 4.0)
    • Android Studio at /home/kapil/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.15.0

[✓] Connected device (1 available)
    • RMX1931 (mobile) • 25.29.66.206:5555 • android-arm64 • Android 10 (API 29)

• No issues found!

**Comment by kapiltewari** [comment]
Closing this because it was the device issue. 
Tested in Nokia 7 plus, release apk works fine, no animation stutter.

**Comment by zippka224** [comment]
> Closing this because it was the device issue.
> Tested in Nokia 7 plus, release apk works fine, no animat

Holy moly, I've just finished developing my simple app and found myself with the same problem. On closer inspection noticed it was X2 Pro , same as my phone....
this sucks... 👎

**Comment by kapiltewari** [comment]
I think its the problem with their Realme UI. There is so much lag or frame drops when playing some high end mobile games too.

**Comment by zippka224** [comment]
I've actually tried LOS 17 and it's actually worse, on relame UI, --profile has no startup stutter but on LOS it is present as well

**Comment by kapiltewari** [comment]
Yes you are right --profile feels really smooth but --release is stuttering.

**Comment by pedromassangocode** [comment]
Cc @liyuqian

**Comment by kapiltewari** [comment]
I'm sorry, that was get closed by mistake.

**Comment by kapiltewari** [comment]
Sorry, this was get closed by mistake.

**Comment by liyuqian** [comment]
This seems to be specific to Xiaomi devices. I just received a Xiaomi 10 and I'll try to reproduce it locally to figure out if there's anything that Flutter can help, or if this is an issue that only Xiaomi can fix.

**Comment by kapiltewari** [comment]
Yes this can be a device specific issue because release APK works fine on Nokia. And the device I am testing in is a Realme device not Xiaomi.

**Comment by kapiltewari** [comment]
But the 'flutter run --profile' works fine, No stutter.

**Comment by zippka224** [comment]
@liyuqian 
Actually, it is not Xiaomi issue. I've tried running the app on my old Redmi Note 5 Pro and it is working as intended. Seems to be Realme issue. 
PS: I have not yet tested this but, it might be R8 compressing which is applied in --release that is causing the initial stuttering.
Need to test this though.

**Comment by liyuqian** [comment]
Ah, I mistakenly thought that Realme is a sub-brand of Xiaomi like Redmi. We don't have a Realme device in our library. CC @kf6gpe who keeps track of our hardware-specific needs. Considering the current resource limit and shipping latency, it may take a long time before we can act on this with a Realme device.

**Comment by kf6gpe** [comment]
Which Realme, and I'll see if I can get one, @liyuqian and @zippka224? May take a week or so for me to get it and get it shipped over to @liyuqian though.

**Comment by zippka224** [comment]
@kf6gpe Realme X2 Pro,
I am running latest OTA Android 10

**Comment by kf6gpe** [comment]
Self-assigning so I remember to track down a handset, which I didn't do before. Stay tuned.

**Comment by kf6gpe** [comment]
Handset shipped to @liyuqian today. Sorry for the delay on my side.

**Comment by RuslanBarcho** [comment]
+1. I facing now same issue with my app. In release build there are visible stutters on every launch, profile also have the same, but less visible and not on every launch.
Only debug build have no stutters at app start. Don't know what is the reason of this problem, but hope it's framework-related one and will be fixed ☹️

**Comment by Hixie** [comment]
@RuslanBarcho Please file a separate bug for that issue, this one is specifically about the Realme phones.

**Comment by Hixie** [comment]
I tried breaking down the GIF in the original report using https://ezgif.com/split to see the stutter but actually it doesn't seem bad at all? I'm not sure I'm seeing what I'm supposed to be seeing. Do you have a video recording by any chance?

**Comment by kapiltewari** [comment]
Sorry, don't have any video recording. If flutter team has Realme X2 Pro by any chance, then this issue can be seen with any app's release apk.

Note that Realme X2 Pro is upgraded to Android 11 since this issue was opened. Maybe they have fixed it. I don't know.

**Comment by darshankawar** [comment]
@kpltwr Is there a way for you to check this issue again using latest stable version (3.38.4) on same affected device to know if the reported behavior still persist or not ? We don't have the said device so we are unable to replicate / verify at our end.
